Xcode 16-beta compilation error: non-private labels cannot appear between .cfi_startproc / .cfi_endproc pairs

I was doing a little bit of scouting on macOS 16 / Xcode 16 and ran in to a compiler error on Xcode 16.

The gist of the error is

```plain
  <instantiation>:5:1: error: non-private labels cannot appear between .cfi_startproc / .cfi_endproc pairs
```

Repeated many times. The full build log is available at https://gist.github.com/vcsjones/1649f2fb8b19ab86428582b6884229c0

Note this does not require a beta of macOS, only the developer tools.

I can reproduce. At first sight, it seems that using `.alt_entry` should be possible (https://github.com/llvm/llvm-project/pull/82268):

```diff
--- a/src/coreclr/nativeaot/Runtime/unix/unixasmmacrosarm64.inc
+++ b/src/coreclr/nativeaot/Runtime/unix/unixasmmacrosarm64.inc
@@ -24,9 +24,11 @@ C_FUNC(\Name):
 .endm
 
 .macro ALTERNATE_ENTRY Name
-        .global C_FUNC(\Name)
 #if !defined(__APPLE__)
+        .global C_FUNC(\Name)
         .hidden C_FUNC(\Name)
+#else
+        .alt_entry C_FUNC(\Name)
 #endif
 C_FUNC(\Name):
 .endm
```

It still fails with the following errors:
```
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/PInvoke.S.o
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_restore fp
          ^
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_restore fp
          ^
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_adjust_cfa_offset -0x60
          ^
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_adjust_cfa_offset -0x40
          ^
  make[3]: *** [nativeaot/Runtime/Full/CMakeFiles/Runtime.WorkstationGC.dir/__/arm64/UniversalTransition.S.o] Error 1
  make[3]: *** Waiting for unfinished jobs....
  make[3]: *** [nativeaot/Runtime/Full/CMakeFiles/Runtime.WorkstationGC.dir/__/arm64/ExceptionHandling.S.o] Error 1
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_adjust_cfa_offset -0x60
          ^
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_adjust_cfa_offset -0x40
          ^
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/InteropThunksHelpers.S.o
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/UniversalTransition.S.o
  make[3]: *** [nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/ExceptionHandling.S.o] Error 1
  make[3]: *** Waiting for unfinished jobs....
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/StubDispatch.S.o
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/WriteBarriers.S.o
  [ 37%] Building ASM object nativeaot/Runtime/Full/CMakeFiles/Runtime.ServerGC.dir/__/arm64/ThunkPoolThunks.S.o
  make[2]: *** [nativeaot/Runtime/Full/CMakeFiles/Runtime.WorkstationGC.dir/all] Error 2
  make[2]: *** Waiting for unfinished jobs....
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_restore fp
          ^
  <instantiation>:2:9: error: invalid CFI advance_loc expression
          .cfi_restore fp
          ^
```

Hopefully it will be relatively easy to figure it out and the linker won't complain about the `.alt_entry`.

--- a/src/coreclr/vm/arm64/asmhelpers.S
+++ b/src/coreclr/vm/arm64/asmhelpers.S
@@ -606,8 +606,12 @@ NESTED_END ResolveWorkerAsmStub, _TEXT
 #ifdef FEATURE_READYTORUN
 
 NESTED_ENTRY DelayLoad_MethodCall_FakeProlog, _TEXT, NoHandler
-C_FUNC(DelayLoad_MethodCall):
+#if defined(__APPLE__)
+    .alt_entry C_FUNC(DelayLoad_MethodCall)
+#endif
     .global C_FUNC(DelayLoad_MethodCall)
+C_FUNC(DelayLoad_MethodCall):
+
     PROLOG_WITH_TRANSITION_BLOCK
 
     add x0, sp, #__PWTB_TransitionBlock // pTransitionBlock
@@ -626,8 +630,11 @@ NESTED_END DelayLoad_MethodCall_FakeProlog, _TEXT
 
 .macro DynamicHelper frameFlags, suffix
 NESTED_ENTRY DelayLoad_Helper\suffix\()_FakeProlog, _TEXT, NoHandler
-C_FUNC(DelayLoad_Helper\suffix):
+#if defined(__APPLE__)
+    .alt_entry C_FUNC(DelayLoad_Helper\suffix)
+#endif
     .global C_FUNC(DelayLoad_Helper\suffix)
+C_FUNC(DelayLoad_Helper\suffix):
 
     PROLOG_WITH_TRANSITION_BLOCK
 

